### PR TITLE
Enable SNTP time sync notification callback.

### PR DESCRIPTION
**Pull request by KerryRJ**
Publish the notification as a TimeSyncEvent.
Use Howard Hinnant's date library.

Please let me know if the date library license can be used in the Smooth library.
If #70 is not merged then the relevant files need to be added to the files.cmake.

### CODE REVIEW

**Comment by PerMalmberg** [comment]
Nice contribution.

I love Howard Hinnant's date library, been using it for quite some time, just not in this project. It is MIT licensed so that should not be a problem, but I think it is better to use it as a Git submodule than including it directly in Smooth.

Also, please use 4 spaces, not tabs for indentation. I'm not used to others contributing so I've not written down any guidelines yet. I'll be committing a configuration for [uncrustify](https://github.com/uncrustify/uncrustify) shortly to be used for formatting the code so there's not question on how it should be.

**Comment by PerMalmberg** [comment]
Master branch now has an uncrustify.cfg that matches the desired code style. I stumbled [upon a possible bug](https://github.com/uncrustify/uncrustify/issues/2383) so please keep a look out for undesired changes when running it on modified files.

**Comment by KerryRJ** [comment]
I now have CodeBeautifier running with Uncrustify.

My git skills are lacking. How and where do I place Howard's library as a submodule? Is the whole library going to be pulled down? Do I assume that I need to place the date.h file in files.cmake?

**Comment by PerMalmberg** [comment]
I can help with that this evening. I've just added nlohman's JSON library as a submodule to replace the one previously used in another branch so I can add this one too. It can then be used via `#include <hinnant/date.h>` or similar. Just make sure you open the PR for modifications by maintainers (if you haven't already).

**Comment by KerryRJ** [comment]
I don't understand what this means nor what is expected from me

> Just make sure you open the PR for modifications by maintainers (if you haven't already).

**Comment by PerMalmberg** [comment]
See here: https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by PerMalmberg** [comment]
I'm in the process of updating tests and mock-idf for SNTP into your branch so please hold off any pushes to this PR until that's done.

**Comment by PerMalmberg** [comment]
@KerryRJ Allright, seems the PR was enabled for changes my maintainers. :+1: 

I've just pushed an update to this PR with changes to tests and the SNTP class itself so that it simulates a SNTP-sync also when run under Linux. Please have a look.

I have not included Howard's date.h since no actual code used it. Did you have further changes in mind that do?

File: lib/compiler_options.cmake [hunk 1]
```diff
@@ -22,6 +22,7 @@ function(set_compile_options target)
     target_compile_options(${target} PRIVATE -Wconversion -Wsign-conversion)
     target_compile_options(${target} PRIVATE -Wlogical-op -Wdouble-promotion -Wformat=2)
     target_compile_options(${target} PUBLIC $<$<COMPILE_LANGUAGE:CXX>: -fno-rtti>)
+	target_compile_definitions(${target} PUBLIC HAS_UNCAUGHT_EXCEPTIONS)
```
Comment (KerryRJ) [inline lib/compiler_options.cmake]: This change was only necessary when using date.h.
It can be removed.

**Comment by KerryRJ** [comment]
If I use date.h in any components later then I'll figure out how to add it as a sub-module.

File: lib/compiler_options.cmake (hunk 1 above)
Comment (KerryRJ) [reply lib/compiler_options.cmake]: Should I delete this file?

Comment (PerMalmberg) [reply lib/compiler_options.cmake]: Well, not the file but please go ahead and delete the file.

**Comment by PerMalmberg** [comment]
Ah, cool - you removed the date.h, I had forgotten to do that.

Thanks for the contribution!

**Pull request merged**